pkg/plugin/group/scaler.go

```diff
@@ -242,6 +259,10 @@ func (s *scaler) converge() {
 
 				s.scaled.CreateOne(nil)
 			}()
+			if s.maxParallelNum > 0 && (i+1)%int(s.maxParallelNum) == 0 {
```

Comment:
How about extract this into a function on the scaler:

```
func (s *scaler) waitIfReachParallelLimit(current int, batch sync.WaitGroup) {
   // call wait on the batch if the current index reaches the parallel limit
}
```
and then call this also after line [248](https://github.com/docker/infrakit/pull/332/files#diff-fdb002f7e382abf7f4b1a8a761139830R248) so that we can rate limit the Destroy calls as well?